services/provider/server/server.go

```diff
@@ -530,103 +530,112 @@ func (s *OCSProviderServer) GetStorageClassClaimConfig(ctx context.Context, req
 		return nil, status.Errorf(codes.Internal, "failed to get storage class claim config %q for %q. %v", req.StorageClassClaimName, req.StorageConsumerUUID, err)
 	}
 
-	// TODO: add status.Phase validation here to check claim is in expected
-	// state or not before extracting the ceph resources from the StorageClassClaim CR status
+	// Verify Status.Phase
+	msg := fmt.Sprintf("storage class claim %q for %q is in %q phase", req.StorageClassClaimName, req.StorageConsumerUUID, storageClassClaim.Status.Phase)
+	switch storageClassClaim.Status.Phase {
+	case ocsv1alpha1.StorageClassClaimFailed:
+		return nil, status.Error(codes.Internal, msg)
+	case ocsv1alpha1.StorageClassClaimInitializing:
+		return nil, status.Error(codes.Unavailable, msg)
+	case ocsv1alpha1.StorageClassClaimCreating:
+		return nil, status.Error(codes.Unavailable, msg)
+	case ocsv1alpha1.StorageClassClaimReady:
```

Comment:
This nesting indent the entirety of the code in this function and produce a switch case inside a switch case. 

```suggestion
        if storageClassClaim.Status.Phase != ocsv1alpha1.StorageClassClaimReady {
		switch storageClassClaim.Status.Phase {
		case ocsv1alpha1.StorageClassClaimInitializing:
			return nil, status.Error(codes.Unavailable, msg)
		case ocsv1alpha1.StorageClassClaimCreating:
			return nil, status.Error(codes.Unavailable, msg)
		default:
			return nil, status.Error(codes.Internal, msg)
	}
	
	// All of the main code of the function goes here without additional indentation 
```

Reply:
sounds better :+1: